lib/active_merchant/billing/gateways/payrix.rb

```diff
@@ -0,0 +1,319 @@
+        post[:city] = truncate(options[:city], ADDRESS_MAX_SIZE)
+        post[:state] = truncate(options[:state], STATE_MAX_SIZE)
+        post[:zip] = truncate(scrub_zip(options[:zip]), ZIP_MAX_SIZE)
+        post[:country] = country_code(options[:country])
+      end
+
+      def add_merchant(post, options)
+        post[:merchant] = options[:merchant_id] || @options[:merchant_id]
+      end
+
+      def add_payment(post, payment)
+        post[:payment] = {
+          method: CREDIT_CARD_CODES[:"#{payment.brand}"],
+          number: payment.number,
+          cvv: payment.verification_value
+        }
+      end
+
+      def add_invoice(post, money, options)
+        post[:total] = money
+        post[:currency] = options[:currency] || default_currency
+        post[:order] = truncate(options[:order], ORDER_MAX_SIZE)
+      end
+
+      def add_transaction_details(post, options)
+        post[:type] = options[:type] || @options[:type]
+        post[:origin] = options[:origin] || @options[:origin]
+        post[:expiration] = options[:expiration] || @options[:expiration]
+        post[:description] = truncate(options[:description], DESCRIPTION_MAX_SIZE)
+        post[:fundingCurrency] = options[:fundingCurrency]
+        post[:cofType] = options[:cofType]
+        post[:allowPartial] = options[:allowPartial] || TXNS_ALLOW_PARTIAL[:partial_amount_authorizations_not_allowed]
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def commit(action, parameters, options={})
+        url = (test? ? test_url : live_url)
+        endpoint = url + '/' + action
+        response = parse(ssl_post(endpoint, parameters.to_json, auth_headers(options)))
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      rescue ActiveMerchant::ResponseError => e
+        response = e.response.body.present? ? parse(e.response.body) : { "response" => { "data" => [], "errors" => [{ "msg" => e.response.msg }] } }
+        message = e.response.msg
+        Response.new(false, message, response, test: test?)
+      end
+
+      def auth_headers(options)
+        {
+          'Content-Type' => 'application/json',
+          'APIKEY' => options[:api_key] || @options[:api_key]
+        }
+      end
+
+      def success_from(response)
+        response["response"]["errors"].empty? ? true : false
+      end
+
+      def message_from(response)
+        if success_from(response)
+          response_status_code = response.try(:[], "response").try(:[], "data").first.try(:[], "status")
+          response_status_code.nil? ? 'Request Successful' : TXNS_RESPONSE_STATUS[:"#{response_status_code}"]
+        else
+          response.try(:[], "response").try(:[], "errors").first.try(:[], "msg")
```

Comment:
I *think* you can do response&.dig("response", "errors") here. But what you have is perfect too :)